### ISSUE TITLE
Wrong back button link

### ISSUE THREAD

**Issue by insky** [supertokens/supertokens-auth-react#625]
If I click "Forgot password" and then back button, URL is {websiteDomain}/{websiteBasePath}?show=signin (no trailing slash), but {websiteDomain}/{websiteBasePath}/ (with trailing slash) expected.

**Comment by rishabhpoddar** [comment]
hey @insky why does having a trailing slash (or not) matter?

**Comment by insky** [comment]
@rishabhpoddar because of uniformity of routes. For example, we have {websiteDomain}/{websiteBasePath}/reset-password
Anyway, component doesn't consider have websiteBasePath trailing slash or not.

**Comment by rishabhpoddar** [comment]
Yea.. I don't think this matters too much. Im closing this issue.

**Comment by egorFiNE** [comment]
@rishabhpoddar we have the same issue. Some http routing configurations are unexpectedly broken because of that. We have routed everything `/{websiteBasePath}/` to the backend and then found out that reset password is broken due to the fact that supertokens routes this one specific request to a _different_ albeit very similar url. 

So we had to add additional deciphering logic. 

Before: 

* does the URL begin with `/{websiteBasePath}/`? Yes → route to backend

After we have discovered this issue: 

* does the URL begin with `/{websiteBasePath}/`? Yes → route to backend
* does the URL begin with `/{websiteBasePath}`? 
  * is this the full url? Yes → route to backend
  * is there a slash after? Yes → route to backend 
  * is there a querystring immediately after? Yes → route to backend
  * is there a hash immediately after? Yes → route to backend
  * is this a beginning of a different and valid pathname starting with the same prefix? Yes → route elsewhere
  
All of this to work around supertokens weird quirk with URLs. 

So in fact I'd say that having a consistent and expected url schema in fact does matter.
